src/instances.ts

```diff
@@ -285,10 +285,19 @@ function successfulTypeScriptInstance(
       loader,
       instance
     );
-    instance.solutionBuilder = compiler.createSolutionBuilderWithWatch(
+    instance.solutionBuilder = compiler.createSolutionBuilder(
       instance.solutionBuilderHost,
       [configFilePath],
-      { verbose: true, watch: true }
+      { verbose: true }
+    );
+    loader._compiler.hooks.watchRun.tap(
```

Comment:
I am not sure if this is correct way to do cleanup before watch compilation begins.. Also I think we need to create solutionbuilder host and solutionBuilder both again but I don't know how to check that..

Reply:
Edit i checked this isnt right way to do this. I think we need to create with watch host when running with --watch. Can someone let me know how to detect that here?

Reply:
There should be a `watchMode` property on the webpack `Compiler` object: https://github.com/webpack/webpack/pull/8253/files

Reply:
I think even with that we need to implement `watchFile` as `loader.addDepencedency` but that needs more work I think given there are too many places clearing dependencies...